### PR TITLE
Update submodules and DMD version

### LINKED ISSUE #20: Update to use dmqproto v14.1.x

### LINKED ISSUE #32: Update to latest ocean and swarm majors

### PR DESCRIPTION
**Pull request by david-eckardt-sociomantic**
Fixes #20.
Fixes #32.
Based on #33.
Needs DMD v2.078 in addition, which in turn needs swarm v5.0.5 and dmqproto v14.1.1.

### CODE REVIEW

File: src/dmqnode/node/DmqNode.d [hunk 1]
```diff
@@ -182,4 +192,21 @@ public class DmqNode
...
         return ["pushed", "popped"];
     }
+
+    /***************************************************************************
+
+        Calls `callback` with a `RequestResources` object whose scope is limited
+        to the run-time of `callback`.
+
+        Params:
+            callback = a callback to pass with a a `RequestResources` object
```
Comment (gavin-norman-sociomantic) [inline src/dmqnode/node/DmqNode.d]: Comment seems kind of garbled.

File: integrationtest/loadfiles/cases/checker/CheckedRequests.d [hunk 2]
```diff
@@ -266,6 +266,8 @@ class UnexpectedNotification
 
 class Consume: RecordChecker
 {
+    import dmqproto.client.request.Consume;
```
Comment (gavin-norman-sociomantic) [inline integrationtest/loadfiles/cases/checker/CheckedRequests.d]: Are these changes necessary? The client was designed so that importing request modules in user code would not be necessary.

Comment (nemanja-boric-sociomantic) [reply integrationtest/loadfiles/cases/checker/CheckedRequests.d]: Trick is that this doesn't work with the 2.071 import rules (I think).

Comment (gavin-norman-sociomantic) [reply integrationtest/loadfiles/cases/checker/CheckedRequests.d]: Can we fix this in the client? It would be nice for it to keep working and not require all users to change their code. The required change won't be at all obvious to the average user.

Comment (nemanja-boric-sociomantic) [reply integrationtest/loadfiles/cases/checker/CheckedRequests.d]: I would hope so it's just the matter of making these imports `public`: https://github.com/sociomantic-tsunami/dmqproto/blob/v14.x.x/src/dmqproto/client/mixins/NeoSupport.d#L51

Comment (nemanja-boric-sociomantic) [reply integrationtest/loadfiles/cases/checker/CheckedRequests.d]: I'll checkout this branch and give it a try, I agree this is a very bad thing to push to the users.

Comment (nemanja-boric-sociomantic) [reply integrationtest/loadfiles/cases/checker/CheckedRequests.d]: OK, looks like that fixes it. I'll submit PR to dmqproto.

Without the change imports in `NeoMixin` to public:

```
integrationtest/loadfiles/cases/checker/CheckedRequests.d(352): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Consume is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(353): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Consume is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(385): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Pop is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(422): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Pop is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(423): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Pop is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(502): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Pop is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(503): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Pop is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(352): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Consume is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(353): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Consume is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(385): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Pop is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(422): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Pop is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(423): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Pop is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(502): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Pop is not visible from module CheckedRequests
integrationtest/loadfiles/cases/checker/CheckedRequests.d(503): Deprecation: dmqproto.client.DmqClient.DmqClient.NeoSupport!().Neo.Pop is not visible from module CheckedRequests

```

With the change:

```
   rdmd build/devel/tmp/test-loadfiles
   run build/devel/tmp/test-loadfiles
```

Comment (nemanja-boric-sociomantic) [reply integrationtest/loadfiles/cases/checker/CheckedRequests.d]: https://github.com/sociomantic-tsunami/dmqproto/pull/56

**Comment by gavin-norman-sociomantic** [comment]
Blocked on [dmqproto v14.1.1](https://github.com/sociomantic-tsunami/dmqproto/milestone/11).

**Comment by david-eckardt-sociomantic** [comment]
Now blocked only on dmqproto v14.1.1 to avoid the import deprecation. We'll use DMD v2.071.

**Comment by nemanja-boric-sociomantic** [comment]
Swarm v5.0.6 is released and as we're waiting for the dmqproto v14.1.1, we can turn on the support for v2.078 as well while we're here.

**Comment by gavin-norman-sociomantic** [comment]
[dmqproto v14.1.1](https://github.com/sociomantic-tsunami/dmqproto/releases/tag/v14.1.1) released.

**Comment by david-eckardt-sociomantic** [comment]
Updated to dmqproto v14.1.1 and swarm v5.0.9. Unblocked.

File: .travis.yml [hunk 3]
```diff
@@ -19,10 +19,10 @@ env:
...
         - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
     matrix:
-        - DMD=dmd1 DIST=xenial F=production
-        - DMD=dmd1 DIST=xenial F=devel
-        - DMD=2.070.2.s* DIST=xenial F=production
-        - DMD=2.070.2.s* DIST=xenial F=devel
+        - DMD=1.081.* DIST=xenial F=production
+        - DMD=1.081.* DIST=xenial F=devel
+        - DMD=2.071.2.s* DIST=xenial F=production
+        - DMD=2.071.2.s* DIST=xenial F=devel
```
Comment (nemanja-boric-sociomantic) [inline .travis.yml]: Now you can even add `2.078.*` here.

**Comment by david-eckardt-sociomantic** [comment]
Suddenly the production builds hang running the tests. Reverted beaver/dmd to see if they succeed then.

**Comment by david-eckardt-sociomantic** [comment]
Tests passed twice with beaver v0.31. Using v0.40 again to see if the tests fail.

**Comment by david-eckardt-sociomantic** [comment]
`F=production` tests got stuck twice with beaver v0.40 (same `travis.yml` i.e. unchanged DMD configuration).

**Comment by david-eckardt-sociomantic** [comment]
Tests pass with Beaver v0.3.3.

**Comment by david-eckardt-sociomantic** [comment]
Tests pass with Beaver v0.3.4.

**Comment by david-eckardt-sociomantic** [comment]
Tests hang with Beaver v0.4.0.
I don't think Beaver is the actual cause but the dmqnode test is flaky. Unfortunately I cannot reproduce the hanging test locally.

**Comment by david-eckardt-sociomantic** [comment]
Updated using Beaver v0.3.4 and DMD 2.078 in addition and fixing the code documentation.

**Comment by gavin-norman-sociomantic** [comment]
Cool, let's continue with this PR and investigate the beaver / test hangs separately. Please report any insights here: https://github.com/sociomantic-tsunami/dmqnode/issues/41.

File: src/dmqnode/request/neo/Push.d [hunk 4]
```diff
@@ -13,81 +13,28 @@
...
-                }
-            );
-            break;
-    }
-}
-
 /*******************************************************************************
 
-    DMQ node implementation of the v0 Push request protocol.
+    DMQ node implementation of the v2 Push request protocol.
```
Comment (gavin-norman-sociomantic) [inline src/dmqnode/request/neo/Push.d]: v3

File: integrationtest/loadfiles/cases/checker/CheckedRequests.d (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply integrationtest/loadfiles/cases/checker/CheckedRequests.d]: These request module import changes should be removed now that dmqproto has been fixed.

**Comment by david-eckardt-sociomantic** [comment]
Updated, correcting the request version documentation and reverting extra imports.
